lib/Parse/ParseDecl.cpp

```diff
@@ -5748,7 +5748,7 @@ Parser::parseDeclEnumCase(ParseDeclOptions Flags,
 
     // Handle the likely case someone typed 'case X, case Y'.
     if (Tok.is(tok::kw_case) && CommaLoc.isValid()) {
-      diagnose(Tok, diag::expected_identifier_after_case_comma);
+      diagnose(CommaLoc, diag::expected_identifier_after_case_comma);
```

Comment:
You changed the first argument from `Tok` to `CommaLoc`. The right approach is to use `PointsToFirstBadToken` in `include/swift/AST/DiagnosticsParse.def`. If a diagnostic has `PointsToFirstBadToken` modifier, the diagnostic location becomes "at the end of previous token".

Reply:
Sure, I'll change it :))